pkgs/tools/security/cryptomator/default.nix

```diff
@@ -0,0 +1,91 @@
+}:
+
+let
+  pname = "cryptomator";
+  version = "1.5.12";
+
+  src = fetchFromGitHub {
+    owner = "cryptomator";
+    repo = "cryptomator";
+    rev = version;
+    sha256 = "1v7s3y0dh00vbi4zdlg3msv3034sa5qzx5fq5xwi76vs2r37nk2j";
+  };
+
+  icons = fetchFromGitHub {
+    owner = "cryptomator";
+    repo = "cryptomator-linux";
+    rev = version;
+    sha256 = "1s6rvkvxmbym5lql8hdyg7av102mvpd11w80f1nbwilqndjskwrd";
+  };
+
+  # perform fake build to make a fixed-output derivation out of the files downloaded from maven central (120MB)
+  deps = stdenv.mkDerivation {
+    name = "cryptomator-${version}-deps";
+    inherit src;
+
+    nativeBuildInputs = [ jdk maven ];
+
+    buildPhase = ''
+      cd main
+      while mvn -Prelease package -Dmaven.repo.local=$out/.m2 -Dmaven.wagon.rto=5000; [ $? = 1 ]; do
+        echo "timeout, restart maven to continue downloading"
+      done
+    '';
+
+    # keep only *.{pom,jar,sha1,nbm} and delete all ephemeral files with lastModified timestamps inside
+    installPhase = ''
+      find $out/.m2 -type f -regex '.+\(\.lastUpdated\|resolver-status\.properties\|_remote\.repositories\)' -delete
+      find $out/.m2 -type f -iname '*.pom' -exec sed -i -e 's/\r\+$//' {} \;
+    '';
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = "15gwldn9dc4gxi8ja4lay5qdadvhxry17ggj3cqamk8g38rh97v0";
+  };
+
+in stdenv.mkDerivation rec {
+  inherit pname version src;
+
+  buildPhase = ''
+    cd main
+    mvn -Prelease package --offline -Dmaven.repo.local=$(cp -dpR ${deps}/.m2 ./ && chmod +w -R .m2 && pwd)/.m2
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin/ $out/usr/share/cryptomator/libs/
+
+    cp buildkit/target/libs/* buildkit/target/linux-libs/* $out/usr/share/cryptomator/libs/
+
+    makeWrapper ${jre}/bin/java $out/bin/cryptomator \
+      --add-flags "-classpath '$out/usr/share/cryptomator/libs/*'" \
+      --add-flags "-Dcryptomator.settingsPath='~/.config/Cryptomator/settings.json'" \
+      --add-flags "-Dcryptomator.ipcPortPath='~/.config/Cryptomator/ipcPort.bin'" \
+      --add-flags "-Dcryptomator.logDir='~/.local/share/Cryptomator/logs'" \
+      --add-flags "-Dcryptomator.mountPointsDir='~/.local/share/Cryptomator/mnt'" \
+      --add-flags "-Djdk.gtk.version=3" \
+      --add-flags "-Xss20m" \
+      --add-flags "-Xmx512m" \
+      --add-flags "org.cryptomator.launcher.Cryptomator" \
+      --prefix PATH : "$out/usr/share/cryptomator/libs/:${lib.makeBinPath [ jre glib ]}" \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ fuse ]}" \
+      --set JAVA_HOME "${jre.home}"
+
+    # install desktop entry and icons
+    cp -r ${icons}/resources/appimage/AppDir/usr $out/
+  '';
+
+  nativeBuildInputs = [ autoPatchelfHook maven makeWrapper wrapGAppsHook jdk ];
+  buildInputs = [ fuse packer jre glib ];
+
+  meta = with lib; {
+    description = "Free client-side encryption for your cloud files";
+    homepage = "https://cryptomator.org";
```

Comment:
@AndersonTorres Is there a guideline for this somewhere?

Reply:
No. 

This is a bit of personal taste, but I think the most important info in `meta` is the homepage, since it is a primary source of information.
After this, the other attributes appear in alphabetic order, with longDescription right after description.

Reply:
> This is a bit of personal taste

IMO we should avoid this in reviews because personal tastes differ (and technically the order in a set doesn't matter). Just my 2¢.

Reply:
Technically the order in a set doesn't matter, but this is not a "compiler is not complaining" issue.
After all, the first lines of almost all expressions here start with `{ lib, stdenv, fetchurl, ...}` even if there is no explicit guideline mentioning it in the documentation. It is an unwritten convention, even if the order doesn't matter.

Besides, I argued for this taste and why it is a good advice. Homepage is the most useful information about a piece of software.

On the other hand, I concede I could write it as a suggestion, not in a so impositive tone.

Reply:
> After all, the first lines of almost all expressions here start with { lib, stdenv, fetchurl, ...} even if there is no explicit guideline mentioning it in the documentation.

This is more intuitive though and AFAIK much more common. And I usually leave this up to the package maintainer as long as it looks reasonable (there are much more important things IMO). And with tools like https://nix-community.github.io/nixpkgs-fmt/ we seem to be transitioning towards one attribute per line.

> It is an unwritten convention, even if the order doesn't matter.

I'm not aware of that convention and we should really document them. IIRC my personal convention is to start with `description`. And https://nixos.org/manual/nixpkgs/unstable/  mixes both "conventions".

> Besides, I argued for this taste and why it is a good advice. Homepage is the most useful information about a piece of software.

IMO this should either be a **documented** convention/guideline/whatever or it shouldn't matter. The reason for that is that otherwise this leads to nitpicking and bikeshedding (no offence).
Especially in this case we really need someone to finally merge this PR (or find important issues) as @bachp's time is also valuable and IMO they shouldn't have to deal with one style nitpick after another (especially if they're based on personal taste).
(No hard feelings but I had to clearly state my opinion here as I'm increasingly concerned that we're unnecessarily annoying/burdening contributors. That's a general problem though and with tools like `nixpkgs-fmt` this might soon be resolved.)